### PR TITLE
Add full keybinding args completions for `commands.run`

### PR DESCRIPTION
**Pull request by zardoy**
@usernamehw This is what I mean in https://github.com/usernamehw/vscode-commands/issues/18#issuecomment-1411973557, I have some keybindings in keymap extension and keybindings.json like:

```json
{
  "key": "cmd+\\",
  "command": "commands.run",
  "args": [
    "volar.action.restartServer",
    {
      "command": "workbench.action.quickOpen",
      "args": "view vue sem",
      "delay": 500
    }
  ],
  "when": "editorLangId == 'vue'"
}
```

It adds autocomplete for all this!

### CODE REVIEW

File: src/keybindingsCompletions.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { commands, CompletionItemKind, languages, Range } from 'vscode';
+import { findNodeAtOffset, getNodePath, getNodeValue, Node, parseTree } from 'jsonc-parser';
+import { getAllCommandPaletteCommands } from './quickPick';
+
+export default () => {
+	languages.registerCompletionItemProvider({
+		pattern: '**/{keybindings.json,package.json}',
```
Comment (usernamehw) [inline src/keybindingsCompletions.ts:7]: Does it provide any completions in `package.json` file?

File: src/quickPick.ts [hunk 2]
```diff
@@ -136,7 +136,12 @@ interface VSCodeCommand {
 
 export type VSCodeCommandWithoutCategory = Omit<VSCodeCommand, 'category'>;
 
-async function getAllCommandPaletteCommands(): Promise<VSCodeCommandWithoutCategory[]> {
+let allCommandPaletteCommandsCached: VSCodeCommandWithoutCategory[] | undefined;
```
Comment (usernamehw) [inline src/quickPick.ts:139]: `allCommandPaletteCommandsCached` seems to be duplicating `$state.allCommandPaletteCommands`.

File: src/keybindingsCompletions.ts (hunk 1 above)
Comment (zardoy) [reply src/keybindingsCompletions.ts:7]: yep, this is for keymaps extensions, really easy to handle, see code below https://github.com/usernamehw/vscode-commands/pull/41/files#diff-b598b9f36ef620aa25d5fe06420037be55db10208caf3da0d2c32c1d20630315R27

Example of such extension: https://github.com/zardoy/vscode-keybindings/blob/6dcb739ae8dd46eb28c7afb59ed5df1091de3ef1/package.json#L236

**Comment by usernamehw** [comment]
It looks great! Maybe I can clean up the `package.json` mess https://github.com/usernamehw/vscode-commands/blob/d1aa7d99f50d425439fae8b28fe77bbfb8148778/package.json#L61-L878 by contributing autocomplete for `settings.json` in the future.

**Comment by zardoy** [comment]
> It looks great! Maybe I can clean up the package.json mess

I can ony suggest you to do TS type -> json schema generation at build step with `typescript-json-schema` module, for example I have an [extension](https://github.com/zardoy/vscode-better-snippets) that doesn't have configuration property at all because it gets generated from [typescript type](https://github.com/zardoy/vscode-better-snippets/blob/829a6bde0886a007e1cd7daf9a59ce850e601ea6/src/configurationType.ts#L129) at build. In my case 310 lines of code gets generated into 1660 json loc.

**Comment by usernamehw** [comment]
Yeah, initial schema was generated with that package https://github.com/usernamehw/vscode-commands/blob/3a0063bcaaac8fb401c77c40cfd2a9971d6078a3/scripts/generateJsonSchema.ts#L1-L27

**Comment by usernamehw** [comment]
@zardoy I'm not sure about autocomplete in `package.json`. It seems like a **very** narrow use case. It will parse the `package.json` file on every completion provider invocation. Will remove that.

**Pull request merged**